cmd/skaffold/app/cmd/flags.go

```diff
@@ -548,16 +548,22 @@ func reflectValueOf(values []interface{}) []reflect.Value {
 	return results
 }
 
-func ParseFlags(cmd *cobra.Command, flags []*Flag) {
+func ResetFlagDefaults(cmd *cobra.Command, flags []*Flag) {
 	// Update default values.
 	for _, fl := range flags {
 		flag := cmd.Flag(fl.Name)
-		if fl.DefValuePerCommand != nil {
-			if defValue, present := fl.DefValuePerCommand[cmd.Use]; present {
-				if !flag.Changed {
-					flag.Value.Set(fmt.Sprintf("%v", defValue))
+		if !flag.Changed {
+			defValue := fl.DefValue
+			if fl.DefValuePerCommand != nil {
+				if d, present := fl.DefValuePerCommand[cmd.Use]; present {
+					defValue = d
 				}
 			}
+			if sv, ok := flag.Value.(pflag.SliceValue); ok {
+				reflect.ValueOf(sv).MethodByName("Replace").Call(reflectValueOf([]interface{}{defValue}))
```

Comment:
I like to stay away from reflections as it can easily lead to runtime errors. However, unit tests would cover it in and if we upgrade the cobra library and this method is no longer supportedl